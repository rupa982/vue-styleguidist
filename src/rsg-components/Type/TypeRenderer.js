import React from 'react';
import PropTypes from 'prop-types';
import Code from 'rsg-components/Code';
import Styled from 'rsg-components/Styled';

const styles = ({ fontFamily, fontSize, color }) => ({
	type: {
		fontFamily: fontFamily.monospace,
		fontSize: fontSize.small,
		color: color.type,
	},
});

export function TypeRenderer({ classes, children }) {
	return <Code className={classes.type}>{children}</Code>;
}

TypeRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(TypeRenderer);
