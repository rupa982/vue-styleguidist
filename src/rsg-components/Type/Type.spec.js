import React from 'react';
import { TypeRenderer } from './TypeRenderer';

it('renderer should render type', () => {
	const actual = shallow(<TypeRenderer classes={{}}>Array</TypeRenderer>);

	expect(actual).toMatchSnapshot();
});
