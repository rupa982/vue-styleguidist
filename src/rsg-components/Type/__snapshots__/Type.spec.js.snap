// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render type 1`] = `
<Styled(Code)>
  Array
</Styled(Code)>
`;
